### PR TITLE
Fix the "fresh new hell" HTML comment bug.

### LINKED ISSUE #689: HTML comments on templates are being used as element comments
See the wonder comment here by @notwaldorf https://www.webcomponents.org/element/PolymerElements/paper-input/elements/paper-input

### PR DESCRIPTION
**Pull request by rictic**
Fixes #689
Improves on #674 as we'll do a better job of not picking up on unrelated comments.

 - [x] CHANGELOG.md has been updated

The solution was surprisingly subtle! The core of it is `getPreviousSiblings`.

### CODE REVIEW

File: src/model/inline-document.ts [hunk 1]
```diff
@@ -89,19 +90,80 @@ export class InlineDocument extends Document {
...
   return jsdoc.unindent(comment).trim();
 }
 
+function getAttachedCommentNode(node: ASTNode): ASTNode|undefined {
+  for (const predecessor of getPreviousSiblings(node)) {
+    // Ignore nodes that are effectively invisible to the element author.
+    if (isFakeNode(predecessor)) {
+      continue;
+    }
+    if (dom5.isTextNode(predecessor)) {
```
Comment (justinfagnani) [inline src/model/inline-document.ts]: can you add a comment here too?

File: src/model/inline-document.ts [hunk 2]
```diff
@@ -89,19 +90,80 @@ export class InlineDocument extends Document {
...
+  }
+  const comment = dom5.getTextContent(commentNode);
   if (!comment || /@license/.test(comment)) {
     return;
   }
   return jsdoc.unindent(comment).trim();
 }
 
+function getAttachedCommentNode(node: ASTNode): ASTNode|undefined {
+  for (const predecessor of getPreviousSiblings(node)) {
```
Comment (justinfagnani) [inline src/model/inline-document.ts]: add a comment on the loop

File: src/model/inline-document.ts [hunk 3]
```diff
@@ -89,19 +90,80 @@ export class InlineDocument extends Document {
...
+    // Ignore nodes that are effectively invisible to the element author.
+    if (isFakeNode(predecessor)) {
+      continue;
```
Comment (justinfagnani) [inline src/model/inline-document.ts]: is this case ever hit?

File: src/polymer/polymer-element.ts [hunk 4]
```diff
@@ -257,10 +257,17 @@ export class PolymerElement extends Element implements PolymerExtension {
...
 
     if (domModule) {
-      this.description = this.description || domModule.comment || '';
       this.domModule = domModule.node;
       this.slots = this.slots.concat(domModule.slots);
       this.localIds = domModule.localIds.slice();
+      if (domModule.comment) {
+        const domModuleJsdoc = jsdoc.parseJsdoc(domModule.comment);
+        this.demos = [...jsdoc.extractDemos(domModuleJsdoc), ...this.demos];
+        if (domModuleJsdoc.description) {
```
Comment (justinfagnani) [inline src/polymer/polymer-element.ts]: some comments here would be good too

File: src/model/inline-document.ts (hunk 2 above)
Comment (rictic) [inline src/model/inline-document.ts]: Clarified the implementation.

File: src/model/inline-document.ts (hunk 3 above)
Comment (rictic) [inline src/model/inline-document.ts]: Yes, see comment on `getPreviousSiblings`

File: src/model/inline-document.ts (hunk 1 above)
Comment (rictic) [inline src/model/inline-document.ts]: Done

File: src/polymer/polymer-element.ts (hunk 4 above)
Comment (rictic) [inline src/polymer/polymer-element.ts]: Done

**Comment by arthurevans** [comment]
Can we get a release tagged and deployed to webcomponents.org? @samuelli

**Comment by samuelli** [comment]
@rictic said he was looking into another issue. We'll get this deployed in due time :)

**Pull request merged**